packages/theme/getStore.ts

```diff
@@ -0,0 +1,58 @@
+import axios from 'axios';
+import urlJoin from 'url-join';
+
+interface StoreConfig {
+  backendUrl: string
+}
+
+/* eslint-disable camelcase */
+export interface StoreAttr {
+  type: string
+  id: string
+  attributes: {
+    name: string
+    url: string
+    meta_description: string | null
+    meta_keywords: string | null
+    seo_title: string | null
+    default_currency: string
+    supported_currencies: string
+    default_locale: string
+    supported_locales: string
+  }
+}
+interface IStore {
+  data: StoreAttr
+}
+interface Store {
+  name: string
+  url: string
+  metaDescription?: string
+  metaKeywords?: string
+  seoTitle?: string
+  defaultCurrency: string
+  supportedCurrencies: string
+  defaultLocale: string
+  supportedLocales: string
+}
+
+const apiRoute = '/api/v2/storefront/store';
```

Comment:
I'm still thinking about this tight coupling of the API route. `getStore` returns `Store` based on the given `StoreConfig`. This gives us enough flexibility to provide our custom base-url of the store-api, however, not enough to have custom route, which could be limiting in some edge cases.

I'd rather see passing the API route as one of the `StoreConfig` parameters. We could make it optional with `/api/v2/storefront/store` as a default value.

Reply:
Placing that `apiRoute` in the `nuxt.config` would be also easier to find if someone wanted to modify it. On the other hand, I don't think this endpoint will change since it's specific to how Spree is constructed internally.

I'll move it to `nuxt.config` and we can rethink it later.